portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,150 @@
+    extraContent.style.display = "block"; 
+    //transition the extra content in
+    extraContent.style.opacity = 0;
+    setTimeout(() => {
+        extraContent.style.opacity = 1;
+    }, this.animationDelay + 20); 
+    secretText.textContent = "Back"; 
 }
+
+function deactivateEerieMode() {
+    eerieModeOn = false; 
+    mePic.classList.remove("fade-image-out"); 
+    document.body.style.backgroundColor = "white"; 
+    root.style.setProperty('--target-color', "black");
+    root.style.setProperty('--circle-cursor-color', "black"); 
+    root.style.setProperty('--highlighted-text-color', "white"); 
+    mainHeading.classList.add("eerie-text"); 
+    sections.forEach(section => {
+         section.classList.remove("eerie-text"); 
+    });
+    mainHeading.textContent = "Kira Toal's Profile"; 
+    mainText.textContent = "Wasn't that fun?"; 
+    mainText.classList.remove("eerie-text"); 
+    secretText.textContent = "???"; 
+    //the following lines disable extra,  eerie-mode-exclusive features 
+    extraContent.style.display = "none"; 
+}
+
+/**
+ * Animates the mouse as it interacts with section headers. 
+ */
+const sections = document.querySelectorAll(".sections li");
+sections.forEach(section => {
+    section.addEventListener("mouseover", e => {
+        circleAroundMouse.classList.add("grow-cursor");
+        if (eerieModeOn) { section.classList.remove("eerie-text"); } 
+        section.classList.add("color-text");  
+    }); 
+    section.addEventListener("mouseleave", e => {
+        circleAroundMouse.classList.remove("grow-cursor");
+        if (eerieModeOn) { section.classList.add("eerie-text"); } 
+        section.classList.remove("color-text"); 
+    }); 
+    section.addEventListener("click", e => {
+        if (section.id === "secret-text") {
+            mainText.textContent = SECRETMESSAGES[messageIndex]; 
+            if (messageIndex < SECRETMESSAGES.length) {
+                messageIndex += 1; 
+            } else {
+                if (!eerieModeOn) { activateEerieMode(); }
+                else {deactivateEerieMode(); }
+            }
+        } else {
+            if (eerieModeOn) {
+                //reverse the text 
+                let revTextArray = section.dataset.message.split(""); 
+                revTextArray = revTextArray.reverse(); 
+                let revTextString = revTextArray.join(""); 
+                mainText.textContent = revTextString;
+            }
+            else {
+                mainText.textContent = section.dataset.message;
+            } 
+        }
+    })
+});
+
+/**
+ * Recommends a movie based on dropdown input 
+ */
+
+const SPN = ["Hereditary", "It Follows", "The Exorcist", "Ju-On: The Grudge", "Ringu"]
+const NONSPN = ["Midsommar", "The Shining", "Hush", "The Strangers", "Orphan"]
+
+const dropdown = document.querySelector("#subgenres"); 
+let userSelectedValue; 
+const result = document.querySelector("#result");
```

Comment:
These variables are only used in the `recommendMovie` function, so can be locally declared inside the function.

Reply:
Got it, thank you